paddle/fluid/framework/operator.h

```diff
@@ -209,6 +214,12 @@ class ExecutionContext {
     return var == nullptr ? nullptr : &var->Get<T>();
   }
 
+  template <typename T>
+  T* MutableInput(const std::string& name) const {
```

Comment:
Input should be const.  Keep the input not changed directly, you can make the var that needs to change the outputs of this op.